store/db.go

```diff
@@ -29,9 +30,36 @@ const (
 	DbFilename = "ql.db"
 )
 
+// dbLock is used to guarantee both thread-safety and process-safety
+// for db access.
+type dbLock struct {
+	// Mutex here is not necessary in fact, this is just to
+	// please the race detector.
+	sync.Mutex
+	fl *lock.FileLock
+}
+
+func newDBLock(dirPath string) (*dbLock, error) {
+	l, err := lock.NewLock(dirPath, lock.Dir)
+	if err != nil {
+		return nil, err
+	}
+	return &dbLock{fl: l}, nil
+}
+
+func (dl *dbLock) lock() error {
+	dl.Lock()
+	return dl.fl.ExclusiveLock()
```

Comment:
Should we `dl.Unlock()` if this fails?
similarly, in unlock(), should we only Unlock() if the fl.Unlock() succeeds?